core/src/test/java/se/tre/freki/labels/LabelClientTypeContextTest.java

```diff
@@ -152,4 +155,67 @@ public void createIdPublishesEventOnSuccess() throws Exception {
     typeContext.createId("foo").get();
     verify(eventBus).post(any(LabelCreatedEvent.class));
   }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testRenameNewNameExists() throws Exception {
+    typeContext = new LabelClientTypeContext(store, LabelType.METRIC, metrics, eventBus,
+        MAX_CACHE_SIZE);
+    // Can not rename to a name that is already taken
+    typeContext.createId("newName");
+    typeContext.rename("oldName", "newName");
```

Comment:
This test will probably pass even though it should not. You need to block by calling `#get` on the result of both `#createId` and `#rename`.